modules/adotBidAdapter.js

```diff
@@ -0,0 +1,636 @@
+
+  return {
+    markup: shouldUseAdMarkup ? bid.adm : null,
+    markupUrl: !shouldUseAdMarkup ? bid.nurl : null,
+    renderer: getRendererFromBid(bid, impressionData)
+  };
+}
+
+function getRendererFromBid(bid, impressionData) {
+  const isOutstreamImpression = isBidAVideo(bid) &&
+    isImpressionAVideo(impressionData.openRTB) &&
+    isOutstream(impressionData.openRTB.video);
+
+  return isOutstreamImpression
+    ? buildOutstreamRenderer(impressionData)
+    : null;
+}
+
+function buildOutstreamRenderer(impressionData) {
+  const renderer = Renderer.install({
+    url: OUTSTREAM_VIDEO_PLAYER_URL,
+    loaded: false,
+    adUnitCode: impressionData.internal.adUnitCode
+  });
+
+  renderer.setRender((ad) => {
+    getVastXml(ad, (err, vastXml) => {
+      if (err) return;
+
+      ad.renderer.push(() => {
+        const container = impressionData.internal.container
+          ? document.querySelector(impressionData.internal.container)
+          : document.getElementById(impressionData.internal.adUnitCode);
+
+        const player = new window.VASTPlayer(container);
+
+        return player
+          .loadXml(vastXml)
+          .then(() => {
+            player.adVolume = 0;
+          })
+          .then(() => player.startAd())
+          .catch(err => console.error(err));
+      });
+    });
+  });
+
+  return renderer;
+}
+
+function tryParse(data) {
+  try {
+    return JSON.parse(data);
+  } catch (err) {
+    console.error(err);
+    return null;
+  }
+}
+
+const adotBidderSpec = {
+  code: 'adot',
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+  isBidRequestValid(adUnit) {
+    const allowedBidderCodes = [this.code];
+
+    return isObject(adUnit) &&
+      allowedBidderCodes.includes(adUnit.bidder) &&
+      isString(adUnit.adUnitCode) &&
+      isString(adUnit.bidderRequestId) &&
+      isString(adUnit.bidId) &&
+      validateMediaTypes(adUnit.mediaTypes, this.supportedMediaTypes) &&
+      validateParameters(adUnit.params, adUnit);
+  },
+  buildRequests(adUnits, adUnitContext) {
+    if (!adUnits) return null;
+
+    return groupBy(adUnits, 'bidderRequestId').map(group => {
+      const bidRequestId = group.id;
+      const adUnits = groupBy(group.values, 'bidId').map((group) => {
+        const length = group.values.length;
+        return length > 0 && group.values[length - 1]
+      });
+
+      return createServerRequestFromAdUnits(adUnits, bidRequestId, adUnitContext)
+    });
+  },
+  interpretResponse(serverResponse, serverRequest) {
+    if (!validateServerRequest(serverRequest)) return [];
+    if (!validateServerResponse(serverResponse)) return [];
+
+    const bidResponse = serverResponse.body;
+
+    return bidResponse.seatbid
+      .filter(seatBid => isObject(seatBid) && isArray(seatBid.bid))
+      .flatMap(seatBid => seatBidsToAds(seatBid, bidResponse, serverRequest));
+  },
+  getUserSyncs(syncOptions, serverResponses) {},
+  onTimeout(timeoutData) {},
+  onBidWon(ad) {},
+  onSetTargeting(ad) {}
```

Comment:
If you don't plan to use these functions as part of your adapter, you can remove them.  They don't have to be defined.